core/spec.md

```diff
@@ -1021,41 +1036,45 @@ Regardless of how the model is retrieved, the overall format is as follows:
 
 ``` text
 {
-  "schemas": [ "STRING" * ], ?         # available schema formats
+  "schemas": [ "STRING" * ], ?         # Available schema formats
   "attributes": {                      # Registry level extensions
     "STRING": {                        # Attribute name
+      "name": "STRING",                # Same as attribute's key
       "type": "TYPE",                  # bool, string, array, object, ...
       "description": "STRING",
       "enum": [ VALUE * ], ?           # Array of values of type "TYPE"
       "strict": BOOL, ?                # Just "enum" values or not. Def=true
       "required": BOOL, ?              # Default: false
       "attributes": { ... }, ?         # Only if "type" == "object". Children
+      "keyType": "TYPE", ?             # Only if "type" == "map"
       "itemType": "TYPE", ?            # Only if "type" == "array"
 
       "ifValue": {                     # Conditional extensions
         VALUE: {
-          "parentAttributes": { ... }  # Siblings to this "attribute"
+          "siblingAttributes": { ... } # Siblings to this "attribute"
         } *
       }
     } *
   },
-  "groups": [
-    { "singular": "STRING",            # eg. "endpoint"
+  "groups": {
+    "STRING": {                        # Plural name, eg. "endpoints"
```

Comment:
Again with the slightly odd location of comments. I'm beginning to think I just don't understand.

Reply:
Massively, massively - it's somewhat shocking just how much clearer that makes it, for me anyway :)